include/rosparam_shortcuts/parameter.h

```diff
@@ -168,49 +173,56 @@ class NodeConfig
 struct ExampleConfig : public NodeConfig
 {
 	ExampleConfig(const rclcpp::Node::SharedPtr& node) : NodeConfig(node) {
+		// Build and declare the parameters
+
 		// declare simple parameters just like that
-		declare(int1_param);
-		declare(int2_param);
-		declare(int3_param);
+		declare(int1_param.describe("This is the first int parameter"));
+		declare(int2_param.describe("This is the second int parameter"));
+		declare(int3_param.describe("This is the third int parameter"));
 
 		// "double_param", add some description and validity checks
-		declare(double_param)
-		    .describe("This is a very important double parameter", "should be greater than 0 and not 7")
-		    .warnIf([](const double& val, std::string& message) {
-			    message = "Value is 7, are you sure this is a good value?";
-			    return val == 7;
-		    })
-		    .errorIf([](const double& val, std::string& message) {
-			    message = "Value must not be negative!";
-			    return val < 0;
-		    });
+		declare(double_param
```

Comment:
so simple... yeah, this would work, I'm just not sure if it's really the most intuitive pattern. But I'd say we stick to this now. Changing the NodeConfig API would not be such a huge deal if we choose to do so later on.